util/linuxfw doesn't compile on GOARCH=arm

Adding GOARCH=arm cross targets to the CI, I hit failures trying to build `gvisor.dev/gvisor/pkg/hostarch`:

```
# gvisor.dev/gvisor/pkg/hostarch
194
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:44:19: undefined: PageSize
195
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:50:18: undefined: PageSize
196
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:68:19: undefined: HugePageSize
197
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:74:18: undefined: HugePageSize
198
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:81:25: undefined: PageSize
199
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:116:15: undefined: PageSize
200
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:122:27: undefined: PageSize
201
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:134:22: undefined: PageSize
202
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:139:21: undefined: CacheLineSize
203
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:145:38: undefined: CacheLineSize
204
../../../go/pkg/mod/gvisor.dev/gvisor@v0.0.0-20221203005347-703fd9b7fbc0/pkg/hostarch/addr.go:145:38: too many errors
```

This is expected, as gvisor only supports amd64 and arm64. But why is it compiling it?

```
$ GOARCH=arm go mod why gvisor.dev/gvisor/pkg/hostarch
tailscale.com/util/linuxfw
gvisor.dev/gvisor/pkg/abi/linux
gvisor.dev/gvisor/pkg/hostarch
```

Looks like the recently added util/linuxfw pulls a gvisor package to get constants and struct declarations from the linux ABI... which in turn pulls a package that only builds on amd64 and arm64. So, I think unfortunately we're going to have to get our ABI definitions from another, more cross-platform place.

cc @andrew-d . Currently only affects `go test ./...`, since nothing imports util/linuxfw yet... But when something else does import it, it'll break the build on a lot of our linux targets.

--- a/util/linuxfw/iptables.go
+++ b/util/linuxfw/iptables.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build linux && !(386 || loong64)
+//go:build linux && !(386 || loong64 || arm || armbe)
 
 package linuxfw
 

--- a/util/linuxfw/linuxfw_struct_linux_test.go
+++ b/util/linuxfw/linuxfw_struct_linux_test.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build linux && !(386 || loong64)
+//go:build linux && !(386 || loong64 || arm || armbe)
 
 package linuxfw
 

--- a/util/linuxfw/linuxfw_unsupported.go
+++ b/util/linuxfw/linuxfw_unsupported.go
@@ -1,10 +1,10 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-// NOTE: linux_386 and linux_loong64 are currently unsupported due to missing
+// NOTE: linux_{386,loong64,arm,armbe} are currently unsupported due to missing
 // support in upstream dependencies.
 
-//go:build !linux || (linux && (386 || loong64))
+//go:build !linux || (linux && (386 || loong64 || arm || armbe))
 
 package linuxfw
 

--- a/util/linuxfw/nftables.go
+++ b/util/linuxfw/nftables.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build linux && !(386 || loong64)
+//go:build linux && !(386 || loong64 || arm || armbe)
 
 package linuxfw
 

--- a/util/linuxfw/nftables_types.go
+++ b/util/linuxfw/nftables_types.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build linux && !(386 || loong64)
+//go:build linux && !(386 || loong64 || arm || armbe)
 
 package linuxfw
 

